### DISCUSSION TITLE
error while downloading requirement. txt

### DISCUSSION THREAD

**Discussion by dunnne** [termux/termux-app#4199] [General]
Hi, i have problem with rust and cargo, and couldn't download the cargo in rust .can someone help me? 

error: subprocess-exited-with-error

  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [17 lines of output]
      error: command failed: 'cargo'

      Caused by:
          No such file or directory (os error 2)

      Stack backtrace:
         0: <unknown>
         1: <unknown>
         2: <unknown>
         3: <unknown>
         4: <unknown>

      Cargo, the Rust package manager, is not installed or is not on PATH.
      This package requires Rust and Cargo to compile extensions. Install it through
      the system's package manager or via https://rustup.rs/

      Checking for Rust toolchain....
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

**Comment by sylirre** [comment]
Install rust with `pkg install rust`. Avoid all other methods such as rustup.
